aicsimageio/readers/glob_reader.py

```diff
@@ -0,0 +1,354 @@
+            # If provided a list of lists
+            if len(channel_names) > 0 and isinstance(channel_names[0], list):
+                # Ensure that the outer list is the number of scenes
+                if len(channel_names) != len(self.scenes):
+                    raise exceptions.ConflictingArgumentsError(
+                        f"Number of channel name lists provided does not match the "
+                        f"number of scenes found in the file. "
+                        f"Number of scenes: {len(self.scenes)}, "
+                        f"Provided channel name lists: {dim_order}"
+                    )
+            self._channel_names = channel_names
+
+        self.chunk_dims = chunk_dims
+
+        for dim in REQUIRED_CHUNK_DIMS:
+            if dim not in self.chunk_dims:
+                self.chunk_dims.append(dim)
+
+        # Safety measure / "feature"
+        self.chunk_dims = [d.upper() for d in self.chunk_dims]
+
+        self._dim_order = dim_order
+        self._channel_names = channel_names
+
+        if single_file_shape is None:
+            with self._fs.open(self._path) as open_resource:
+                with TiffFile(open_resource) as tiff:
+                    if tiff.is_shaped:
+                        self._single_file_shape = tuple(
+                            tiff.shaped_metadata[0]["shape"]
+                        )
+                    else:
+                        self._single_file_shape = tiff.pages[0].shape
+
+        else:
+            self._single_file_shape = single_file_shape
+
+        if len(single_file_dims) != len(self._single_file_shape):
+            raise exceptions.ConflictingArgumentsError(
+                f"Number of single file dimensions does not match the"
+                f"number of dimensions in a test file. "
+                f"Number of dimensions in file: {len(self._single_file_shape)}, "
+                f"Provided number of dimensions: {len(single_file_dims)}."
+            )
+
+        else:
+            self._single_file_dims = list(single_file_dims)
+
+        # Enforce valid image
+        if not self._is_supported_image(self._fs, self._path):
+            raise exceptions.UnsupportedFileFormatError(
+                self.__class__.__name__, self._path
+            )
+
+    @property
+    def scenes(self) -> Tuple[str, ...]:
+        if self._scenes is None:
+            self._scenes = tuple(
+                metadata_utils.generate_ome_image_id(s)
+                for s in range(self._all_files[DimensionNames.Samples].nunique())
+            )
+        return self._scenes
+
+    def _read_delayed(self) -> xr.DataArray:
+
+        scene_files = self._all_files.loc[
+            self._all_files[DimensionNames.Samples] == self.current_scene_index
+        ]
+        scene_files = scene_files.drop(DimensionNames.Samples, axis=1)
+        group_dims = [
+            x for x in scene_files.columns if x not in self.chunk_dims + ["filename"]
```

Comment:
This might be a bit clearer as:
```suggestion
            x for x in scene_files.columns if x not in ["filename", *self.chunk_dims]
```

Reply:
Agree! List expansion is nicer than addition imo.